### PR TITLE
Apply Defaults in the AddColumn node

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
This pr

1) Adds Defaults in the Add Column Node
2) Adds a nullability check in the Update node

### CODE REVIEW

**Comment by zachmu** [comment]
I think this is probably obsoleted by my PR

**Comment by VinaiRachakonda** [comment]
@zachmu Seems like I just have to rebase off of your pr and switch to your more recent version of apply defaults?

File: sql/plan/alter_table.go [hunk 1]
```diff
@@ -145,7 +146,105 @@ func (a *AddColumn) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
...
+	if err != nil {
+		return nil, err
+	}
+
+	return sql.RowsToRowIter(), a.updateRowsWithDefaults(ctx, row)
+}
+
+// updateRowsWithDefaults iterates through an updatable table and applies an update to each row.
+func (a *AddColumn) updateRowsWithDefaults(ctx *sql.Context, row sql.Row) error {
+	updatable, err := getUpdatable(a.Child)
```
Comment (zachmu) [inline sql/plan/alter_table.go:159]: If you're going to do this you need to update the definition of AlterableTable to include UpdatableTable

File: sql/plan/update.go [hunk 2]
```diff
@@ -200,6 +205,16 @@ func applyUpdateExpressions(ctx *sql.Context, updateExprs []sql.Expression, row
 	return prev, nil
 }
 
+func (u *updateIter) validateNullability(row sql.Row, schema sql.Schema) error {
+	for count, col := range schema {
```
Comment (zachmu) [inline sql/plan/update.go:209]: i or idx better than count

File: sql/plan/alter_table.go [hunk 3]
```diff
@@ -145,7 +146,105 @@ func (a *AddColumn) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
...
+		if err != nil {
+			return nil, err
+		}
+		newRow[col], err = tblSch[col].Type.Convert(val)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if secondPass {
```
Comment (zachmu) [inline sql/plan/alter_table.go:236]: secondPass is actually a no-op in this version of the code (Daylon pointed this out to me on my own PR), get rid of it

File: sql/plan/alter_table.go [hunk 4]
```diff
@@ -145,7 +146,105 @@ func (a *AddColumn) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
...
+		}
+
+		err = updater.Update(ctx, r, updatedRow)
+		if err != nil {
+			return err
+		}
+	}
+}
+
+// applyDefaults applies the default values of the given column indices to the given row, and returns a new row with the updated values.
```
Comment (zachmu) [inline sql/plan/alter_table.go:202]: Column needs updating

**Review by zachmu — APPROVED** [review]
Looks good, just a couple comments

**Pull request merged**